### DISCUSSION TITLE
Add custom data to blocks

### DISCUSSION THREAD

**Discussion by teradrive65k** [sulu/sulu#7083] [Q&A]
Hello everybody, I want to add some custom data to my blocks in sulu, but these data is not present in the original XML file. For exemple by using the headless cms bundle I have a response like this:

```
{
        "type": "blockSomething",
        "settings": [
          
        ],
        "headline": "My something",
        "blocks": [
          {
            "type": "something",
            "settings": [
              
            ],
            "headline": "Super title",
            "textZone": "<p>super text</p>",
          }
        ],
        "textZone": "Another text",
      }
}
```
It's fine but I want to add a property "id" in my json for each block , something like this:
```
{
        "type": "blockSomething",
        "settings": [
          
        ],
        "headline": "My something",
        "blocks": [
          {
            "type": "something",
            "settings": [
              
            ],
            "headline": "Super title",
            "textZone": "<p>super text</p>",
            "id": "noAK2QQQjsQ1ID"
          }
        ],
        "textZone": "Another text",
      }
}
```
How can I add this "id" property ?
For now I get the content in the headless controller and then I add the id by hand by looping recursively on all my data, but I find this very heavy. If I can do it at start it will be better. The issue that I don't know how ? Can I get the object somehow and add this value before the json response ? Do I have to save this property in the database ?

Thanks for the help.

**Comment by alexander-schranz** [comment]
Sorry, just stumbled know over this discussion. The only way currently would be that you add a response listener do add such id. Still I want to mention that we are thinking of adding a id which is unique per page to the core in 3.0: https://github.com/sulu/sulu/issues/7846 if you have something to add let us know there.
